crypto/bio/bio_lib.c

```diff
@@ -93,7 +93,8 @@ int BIO_set(BIO *bio, const BIO_METHOD *method)
     bio->references = 1;
     bio->num_read = 0L;
     bio->num_write = 0L;
-    CRYPTO_new_ex_data(CRYPTO_EX_INDEX_BIO, bio, &bio->ex_data);
+    if (!CRYPTO_new_ex_data(CRYPTO_EX_INDEX_BIO, bio, &bio->ex_data))
```

Comment:
Generally speaking, we prefer NULL pointer checks to be like this:

    ptr == NULL

rather than

    !ptr

So expressions like that should be written like this:

``` C
    if (CRYPTO_new_ex_data(CRYPTO_EX_INDEX_BIO, bio, &bio->ex_data) == NULL)
```

That applies in several places of this change.